receiver/azureeventhubreceiver/azure_formatter_impl_test.go

```diff
@@ -0,0 +1,339 @@
+		})
+	}
+
+}
+
+var minimumLogRecord = func() plog.LogRecord {
+	lr := plog.NewLogs().ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+
+	ts, _ := asTimestamp("2022-11-11T04:48:27.6767145Z")
+	lr.SetTimestamp(ts)
+	lr.Attributes().PutStr(azureOperationName, "SecretGet")
+	lr.Attributes().PutStr(azureCategory, "AuditEvent")
+	lr.Attributes().PutStr(cloudProvider, "azure")
+	lr.Attributes().Sort()
+	return lr
+}()
+
+var maximumLogRecord = func() plog.LogRecord {
+	lr := plog.NewLogs().ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+
+	ts, _ := asTimestamp("2022-11-11T04:48:27.6767145Z")
+	lr.SetTimestamp(ts)
+	lr.SetSeverityNumber(plog.SeverityNumberWarn)
+	lr.SetSeverityText("Warning")
+	guid := "607964b6-41a5-4e24-a5db-db7aab3b9b34"
+
+	lr.Attributes().PutStr(azureTenantID, "/TENANT_ID")
+	lr.Attributes().PutStr(azureOperationName, "SecretGet")
+	lr.Attributes().PutStr(azureOperationVersion, "7.0")
+	lr.Attributes().PutStr(azureCategory, "AuditEvent")
+	lr.Attributes().PutStr(azureCorrelationID, guid)
+	lr.Attributes().PutStr(azureResultType, "Success")
+	lr.Attributes().PutStr(azureResultSignature, "Signature")
+	lr.Attributes().PutStr(azureResultDescription, "Description")
+	lr.Attributes().PutInt(azureDuration, 1234)
+	lr.Attributes().PutStr(netSockPeerAddr, "127.0.0.1")
+	lr.Attributes().PutStr(cloudRegion, "ukso")
+	lr.Attributes().PutStr(cloudProvider, "azure")
+
+	lr.Attributes().PutEmptyMap(azureIdentity).PutEmptyMap("claim").PutStr("oid", "607964b6-41a5-4e24-a5db-db7aab3b9b34")
+	m := lr.Attributes().PutEmptyMap(azureProperties)
+	m.PutStr("string", "string")
+	m.PutDouble("int", 429)
+	m.PutDouble("float", 3.14)
+	m.PutBool("bool", false)
+	m.Sort()
+
+	lr.Attributes().Sort()
+
+	return lr
+}()
+
+// sortLogAttributes is a utility function that will sort
+// all the resource and logRecord attributes to allow
+// reliable comparison in the tests
+func sortLogAttributes(logs plog.Logs) plog.Logs {
+	for r := 0; r < logs.ResourceLogs().Len(); r++ {
+		rl := logs.ResourceLogs().At(r)
+		sortAttributes(rl.Resource().Attributes())
+		for s := 0; s < rl.ScopeLogs().Len(); s++ {
+			sl := rl.ScopeLogs().At(s)
+			for l := 0; l < sl.LogRecords().Len(); l++ {
+				sortAttributes(sl.LogRecords().At(l).Attributes())
+			}
+		}
+	}
+	return logs
+}
+
+// sortAttributes will sort the keys in the given map and
+// then recursively sort any child maps. The function will
+// traverse both maps and slices.
+func sortAttributes(attrs pcommon.Map) {
+	attrs.Range(func(k string, v pcommon.Value) bool {
+		if v.Type() == pcommon.ValueTypeMap {
+			sortAttributes(v.Map())
+		} else if v.Type() == pcommon.ValueTypeSlice {
+			s := v.Slice()
+			for i := 0; i < s.Len(); i++ {
+				value := s.At(i)
+				if value.Type() == pcommon.ValueTypeMap {
+					sortAttributes(value.Map())
+				}
+			}
+		}
+		return true
+	})
+	attrs.Sort()
+}
```

Comment:
if you use `assert.EqualValues`, you shouldn't need to sort maps and their submaps.

Reply:
Using `assert.EqualValues` without sorting fails for me.

Reply:
Mmmm, I believe this might be solved for you. Hold up a minute to see if any other component checks this for you.

Reply:
I believe what you want is `scrapertest.CompareMetrics`, that should be able to compare the expected metric with the metric received :)

Reply:
Thanks for the pointer. However, that particular function only works for metrics, not logs. Looking briefly through the rest of the internal test utilities, I don't see a similar function for logs. I could, of course, pull this out into a general utility, but I'd need guidance on where to put that utility within the internal area.